### DISCUSSION TITLE
Determining layout in custom widget

### DISCUSSION THREAD

**Discussion by 90-008** [hecrj/iced#777] [Q&A]
I'm trying to build a markdown rendering widget with initial goals being rendering code and hyperlinks as seperate, and handling clicks to hyperlinks. I'm using my own markdown parser for parsing (https://github.com/yusdacra/linemd). The issue I'm having with is determining the layout. Basically, my parser produces some tokens which also include line break tokens. I believe each token will need to be it's own node so that in `draw` I can render them stylized. So how could I go about determining the layout for these tokens?

**Comment by 13r0ck** [comment]
Not sure if it helps or if you are already aware, there was a discussion on the Zulip about making a markdown widget late last month. From what I can tell it is on hold, the last mesasge is from Hecrj saying

> #156 is blocked by a text rework. I know it's been pending for a while, but I plan to get to it soon :bow: It's a huge undertaking, so I am making sure it's the right time to fully focus on it. Will share progress here once I start it in case anyone wants to join and help me.

**Reply by dhardy** [reply]
Not surprised; text is hard. I've made a bit more progress on [Markdown formatting for KAS](https://github.com/kas-gui/kas/blob/work/kas-wgpu/examples/README.md#markdown), but it's still woefully incomplete.
